src/object/notification.c

```diff
@@ -187,6 +184,10 @@ void receiveSignal(tcb_t *thread, cap_t cap, bool_t isBlocking)
 
             notification_ptr_set_state(ntfnPtr, NtfnState_Waiting);
             ntfn_ptr_set_queue(ntfnPtr, ntfn_queue);
+
+#ifdef CONFIG_KERNEL_MCS
+            maybeReturnSchedContext(ntfnPtr, thread);
+#endif
```

Comment:
I think the need for the change is the fact that this can move an SC and its easier to show that the invariants still hold after the updates to the notification queue.